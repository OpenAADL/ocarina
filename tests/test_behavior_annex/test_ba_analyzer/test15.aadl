
package BA_example public
with Base_Types;

thread merger
  features
    p1 : in event data port Base_Types::Integer;
    p2 : in event data port Base_Types::Integer;
    m : out event data port Base_Types::Integer;
end merger;

thread implementation merger.impl 
annex behavior_specification {**
  states
    s0 : initial complete state;
    comp : state;
    next1, next2 : complete final state;
  
  transitions
    s0 -[ on dispatch p ]-> next2 { x1 := p1 }; 
    -- errors :
    -- "p" (dispatch trigger) does not point to any event port or subprogram access feature
    s0 -[ on dispatch p2 ]-> next1 { x2 := p2 };
**};
end merger.impl;


end BA_example;
