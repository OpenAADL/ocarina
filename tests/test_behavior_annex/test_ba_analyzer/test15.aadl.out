test15.aadl:20:23: "p" (dispatch trigger) does not point to any event port or subprogram access feature
Cannot analyze BEHAVIOR_SPECIFICATION specifications
